modules/vstudio/tests/cs2005/test_documentation_file.lua

```diff
@@ -0,0 +1,88 @@
+--
+-- tests/actions/vstudio/cs2005/test_documentation_file.lua
+-- Test DocumentationFile feature Visual Studio 2005+ C# project.
+-- Copyright (c) 2012-2023 Jason Perkins and the Premake project
+--
+	local p = premake
+	local suite = test.declare("vstudio_cs2005_documentation_file")
+	local dn2005 = p.vstudio.dotnetbase
+--
+-- Setup
+--
+
+	local wks, prj
+
+--
+-- Setup and teardown
+--
+	function suite.setup()
+		p.action.set("vs2010")
+		wks = test.createWorkspace()
+	 	configurations { "Debug", "Release" }
+		language "C#"
+		targetdir("test\\targetDir")
+	end
+
+	local function setConfig()
+		local cfg = test.getconfig(prj, "Debug")
+		dn2005.documentationfile(cfg);
+	end
+
+
+
+	local function prepare()
+		prj = test.getproject(wks, 1)
+	end
+
+	local function prepareEmpty()
+		prepare()
+		documentationfile ""
+		setConfig()
+	end
+
+	local function prepareDir()
+		prepare()
+		documentationfile "test"
+		setConfig()
+	end
+
+	local function prepareNull()
+		p.action.set("vs2010")
+		wks = test.createWorkspace()
+		setConfig()
+	end
```

Comment:
Not really sure where you got this pattern from, but I haven't seen this in the other unit tests. They generally have a single `prepare`, if there's more than one it's for testing different functions, not testing different values of an API. Please remove these other functions and just use the single `prepare` function pattern. It makes the tests much easier to understand.

Reply:
will do